var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -53,50 +80,103 @@ class Babelstream(CMakePackage, CudaPackage, ROCmPackage):
         when="+cuda",
         msg="CUDA requires architecture to be specfied by cuda_arch=",
     )
-    variant("mem", values=str, default="DEFAULT", description="Enable MEM Target for CUDA")
-    # Raja Conflict
     variant(
-        "offload", values=str, default="none", description="Enable RAJA Target [CPU or NVIDIA]"
-    )
-    conflicts(
-        "offload=none",
+        "cuda_memory_mode",
+        values=("default","managed","pagefault"),
+        default="default",
+        when="+cuda",
+        description="Enable MEM Target for CUDA",
+        )
+
+    # Raja offload
+    variant(
+        "raja_offload",
+        values=("cpu", "nvidia"),
+        default="cpu",
         when="+raja",
-        msg="RAJA requires architecture to be specfied by target=[CPU,NVIDIA]",
+        description="Enable RAJA Target [CPU or NVIDIA] / Offload with custom settings for OpenMP",
+    )
+    #std-* offload
+    variant(
+        "std_offload",
+        values=("nvhpc","none"),
+        default="none",
+        when="+std",
+        description="Enable offloading support (via the non-standard `-stdpar`) for the new NVHPC SDK",
+    )
+    variant(
+        "std_onedpl_backend",
+        values=("openmp","tbb","dpcpp","none"),
+        default="none",
+        when="+std",
+        description="Implements policies using OpenMP,TBB or dpc++",
+    )
+    variant(
+        "std_use_tbb",
+        values=(True,False),
+        default=False,
+        when="+std",
+        description="No-op if ONE_TBB_DIR is set. Link against an in-tree oneTBB via FetchContent_Declare, see top level CMakeLists.txt for details"
+    )
+    variant(
+        "std_use_onedpl",
+        values=(True,False),
+        default=False,
+        when="+std",
+        description="Link oneDPL which implements C++17 executor policies (via execution_policy_tag) for different backends"
+    )
+    #hip memory mode
+    variant(
+        "hip_mem_mode",
+        values=("default","managed","pagefault"),
+        default="default",
+        when="+hip",
+        description="Enable MEM Target for HIP",
+        )
+    #tbb use vector
+    variant(
+        "tbb_use_vector",
+        values=(True,False),
+        default=False,
+        when="+tbb",
+        description="Whether to use std::vector<T> for storage or use aligned_alloc. C++ vectors are *zero* initialised where as aligned_alloc is uninitialised before first use."
     )
-
     # download raja from https://github.com/LLNL/RAJA
     conflicts(
         "dir=none",
         when="+raja",
         msg="RAJA implementation requires architecture to be specfied by dir=",
     )
-
+    # depends_on("cuda@12.3.0:", when="+thrust")
     # Thrust Conflict
     # conflicts("~cuda", when="+thrust", msg="Thrust requires +cuda variant")
     depends_on("thrust", when="+thrust")
-    depends_on("rocthrust", when="+thrust implementation=rocm")
-
+    depends_on("cuda", when="+thrust")
+    depends_on("cuda", when="+raja")
```

Comment:
```suggestion
    depends_on("cuda", when="+raja raja_offload=NVIDIA")
```
Is this a direct dependency, or is it handled entirely through raja? My first thought is that it should be `depends_on("raja+cuda", ...)` and it's on raja to handle the cuda dep.